az-avd_test.go

```diff
@@ -35,12 +37,16 @@ var (
 )
 
 func initAVD() error {
-	_ = testutil.LoadEnv("test.env")
-	tenantId = cloudy.ForceEnv("AZ_TENANT_ID", "")
-	clientId = cloudy.ForceEnv("AZ_CLIENT_ID", "")
-	clientSecret = cloudy.ForceEnv("AZ_CLIENT_SECRET", "")
-	subscriptionId = cloudy.ForceEnv("AZ_SUBSCRIPTION_ID", "")
-	vaultUrl = cloudy.ForceEnv("AZ_VAULT_URL", "")
+	_ = testutil.LoadEnv("../arkloud-conf/arkloud.env")
```

Comment:
Where does this come from?  It isn't in Cloudy.  Needs a comment at least.

Reply:
The combo
>	_ = testutil.LoadEnv("../arkloud-conf/arkloud.env")
>	env := cloudy.CreateCompleteEnvironment("ARKLOUD_ENV", "USERAPI_PREFIX", "KEYVAULT")

... was the convention that @dyerz  suggested using when fixing the unit tests in cloudy-msgraph.  In that case it only loaded the one ../arkloud-conf/arkcloud.env file, but looking at testutil.LoadEnv it doesn't appear that all of the env vars have to be saved in one file.  It may make sense to have multiple env files depending on what's being tested .

Maybe one env that all of the tests share that configures the keyvault env, but then each cloudy-* project could have their own conf files that that just hold env vars that are particular to whatever they are testing.

The original arkloud.env I was given just had the following
> KEYVAULT_AZ_CLIENT_SECRET=
KEYVAULT_AZ_CLIENT_ID=
KEYVAULT_AZ_TENANT_ID=
AZ_VAULT_URL=
ARKLOUD_ENV=test|osenv|azure-keyvault
USERAPI_PREFIX=
DATATYPE_USER=
DATATYPE_PASSWORD=
DATATYPE_HOST=
DATATYPE_DATABASE=

To fix the tests in cloudy-msgraph, these were added

> TEST_USER=
TEST_SKU=

Then to support cloudy-azure, these were added

>ACCOUNT_BLOB=
ACCOUNT_BLOB_KEY=
FS_ACCOUNT=
RG_FILESHARE=

It seems to me that maybe instead of a single loadEnv conf file, there need to be multiple ones - one for the general/keyvault config, and the one for each test suite (or even test case). 

Does that sound right or am I overcomplicating things?

Reply:
We should probably discuss as a team.  My opinion is there should be one per repo.